tests/unit/string_lib_test.c

```diff
@@ -496,6 +496,72 @@ static void test_string_to_long_compatibility(void)
     // Old function (StringToLongUnsafe) does not allow trailing whitespace
 }
 
+#define ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG(str, expected) \
+    assert_int_equal(StringDecimalToLong(str, &value_out), 0); \
+    assert_int_equal(value_out, expected)
+
+static void test_string_decimal_to_long(void)
+{
+    long value_out;
+
+    // Basic usage:
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("0.0", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("-0.0", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("+0.0", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("123.0", 123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("+123.0", 123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("-123.0", (long) -123);
+
+    // Leading spaces:
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG(" 0.0", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG(" 123.0", 123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG(" -123.0", (long) -123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("           0.0", 0);  // 12 is max accepted int length
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("      123.0", 123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("      -123.0", (long) -123);
+
+    // Trailing spaces:
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("9.0 ", 9);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("789.0 ", 789);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("-789.0 ", (long) -789);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("9.0           ", 9);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("789.0         ", 789);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("-789.0        ", (long) -789);
+
+    // More spaces:
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("   0.0     ", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("   -0.0    ", 0);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("   123.0  ", 123);
+
+    // Space separated numbers:
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("   123.456 78   9  ", 123);
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("123.0 45", 123);
+}
+
+static void test_string_decimal_to_long_errors(void)
+{
+    // Successful call returns 0
+    long value_out;
+
+    ASSERT_INT_EQUAL_STRING_DECIMAL_TO_LONG("1234.5678", 1234);
+
+    assert_int_not_equal(StringDecimalToLong("", &value_out), 0);
+    assert_int_not_equal(StringDecimalToLong(" ", &value_out), 0);
+    assert_int_not_equal(StringDecimalToLong("error", &value_out), 0);
+    assert_int_not_equal(StringDecimalToLong("-error", &value_out), 0);
+    assert_int_not_equal(StringDecimalToLong("ffff", &value_out), 0);
+    assert_int_not_equal(StringDecimalToLong("123.456d", &value_out), 0);
```

Comment:
@KotzaBoss I think it is fine to not do any validation past the `.`. If needed we can have a separate function to validate if something is a floating point number.
```suggestion
```

Reply:
Maybe @vpodzime has an opinion on this.